metabase/metabase
# Fix excessive series order settings crash static cartesian chart

Closes https://github.com/metabase/metabase/issues/45397

### Description

When series order settings did not match the actual chart series, for example, due to filterting out some of breakout series, the static version crashed. This is caused by a mistake in the static chart settings computation which preferred the stored `graph.series_order` setting value. However, the settings definition for the dynamic chart uses `getValue` [to always compute it](https://github.com/metabase/metabase/compare/fix-series-order-settings-crash-static-cartesian-chart?expand=1#diff-a996e07afd884cc1340bdeffa0eb577d7828cf4e6418c547d4dc201d1511f96dR128).

Another bug that this PR fixes is the series and legend order in static viz which has become visible after fixing the original issue. This is why so many snapshots got updated. Now series and legend order in static viz fully matches the order in the dynamic viz. Here is the demo:

https://github.com/user-attachments/assets/b31efddc-5acb-4271-992c-2564062af51c

### How to verify

1. New: Orders -> Group by Created at with Product -> Category breakout
2. Reorder series and save the question
3. Now add a filter by Product -> Category and keep only two values out of four
4. Save the question
5. Try to send a dashboard subscription with the question and ensure it works

### Checklist
